### PR TITLE
 !@field filter fails when field does not exist fix #1163  

### LINKED ISSUE #1163: !@field filter fails when field does not exist

### PR DESCRIPTION
**Pull request by ashtul**
fix #1163

### CODE REVIEW

File: src/pytest/test.py [hunk 1]
```diff
@@ -2855,8 +2855,8 @@ def testErrorOnOpperation(env):
     err = env.cmd('ft.aggregate', 'idx', '@test:[0..inf]', 'LOAD', '1', '@test', 'APPLY', '!(split("bad" + "bad"))', 'as', 'a')[1]
     assertEqualIgnoreCluster(env, type(err[0]), redis.exceptions.ResponseError)
 
-    err = env.cmd('ft.aggregate', 'idx', '@test:[0..inf]', 'APPLY', '!@test', 'as', 'a')[1]
-    assertEqualIgnoreCluster(env, type(err[0]), redis.exceptions.ResponseError)
+    #err = env.cmd('ft.aggregate', 'idx', '@test:[0..inf]', 'APPLY', '!@test', 'as', 'a')[1]
```
Comment (MeirShpilraien) [inline src/pytest/test.py:2858]: Removing test is not a good idea, you are breaking behavior (or maybe 1.6 breaks behavior compare to 1.4, we need to talk about it)

File: src/pytest/test_conditional_updates.py [hunk 2]
```diff
@@ -30,11 +30,11 @@ def testConditionalUpdateOnNoneExistingNumericField(env):
     env.assertEqual(env.cmd('ft.add', 'idx', 'doc1', 1.0,
                               'REPLACE', 'PARTIAL',
                               'IF', '@id1 != @id2',
-                              'fields', 'id1', '3', 'id2', '4'), 'NOADD')
+                              'fields', 'id1', '3', 'id2', '4'), 'OK')
```
Comment (MeirShpilraien) [inline src/pytest/test_conditional_updates.py:33]: Same, you break behavior, how does it behave or 1.4?

Comment (ashtul) [reply src/pytest/test_conditional_updates.py:33]: @MeirShpilraien @mnunberg @gkorland 
In 1.4 it behaves the same as before but this seems like wrong behavior.

Comment (MeirShpilraien) [reply src/pytest/test_conditional_updates.py:33]: I agree but changing now is a breaking change

**Comment by MeirShpilraien** [comment]
> The correct fix to this is to ignore `QUERY_ENOPROPVAL` in aggregations; right now it's causing it to throw an error.

Not follow why this is the right solution, we are talking about ft.add with IF statement

**Pull request closed without merging**